### PR TITLE
Space security

### PR DESCRIPTION
**Pull request by BykhovDenis**
# Contribution checklist

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [x] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [x] - Does the translations are up to date?
* [x] - Does it well tested?
* [x] - Tested for Chrome.
* [x] - Tested for Safari.
* [x] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [x] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [x] - Are required links are linked to PR?
* [x] - Does new code is well documented ?

## Related issues
https://front.hc.engineering/workbench/platform/tracker/TSK-714
A list of closed updated issues

### CODE REVIEW

File: server/middleware/src/spaceSecurity.ts [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        await this.filterLookup(ctx, object.$lookup)
+      }
+      res.push(object)
+    }
+    return res
+  }
+
+  async isUnavailable (ctx: SessionContext, space: Ref<Space>): Promise<boolean> {
+    if (this.privateSpaces[space] === undefined) return false
+    if (ctx.userEmail === 'anticrm@hc.engineering') {
```
Comment (haiodo) [inline server/middleware/src/spaceSecurity.ts]: Let's create a ticket to think how to omit this check.

File: server/middleware/src/spaceSecurity.ts [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+  }
+
+  override async findAll<T extends Doc>(
+    ctx: SessionContext,
+    _class: Ref<Class<T>>,
+    query: DocumentQuery<T>,
+    options?: FindOptions<T>
+  ): Promise<FindResult<T>> {
+    const findResult = this.storage.hierarchy.clone(
+      await this.provideFindAll(ctx, _class, query, options)
```
Comment (haiodo) [inline server/middleware/src/spaceSecurity.ts]: Why not {...query, space: {$in: allowed}}?

Comment (BykhovDenis) [reply server/middleware/src/spaceSecurity.ts]: Allowed contains only private spaces

Comment (haiodo) [reply server/middleware/src/spaceSecurity.ts]: I've mean it should be faster to filter on Mongo, not on server. allowed is concept of all spaces allowed.

**Review by haiodo — APPROVED** [review]

**Pull request merged**